cogs/roles.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) idoneam (2016-2019)
+#
+# This file is part of Canary
+#
+# Canary is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Canary is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Canary. If not, see <https://www.gnu.org/licenses/>.
+
+# discord-py requirements
+import discord
+from discord import utils
+from discord.ext import commands
+from .utils.paginator import Pages
+
+
+class Roles(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.roles = self.bot.config.roles
+
+    async def toggle_role(self,
+                          ctx,
+                          transaction,
+                          requested_role,
+                          category,
+                          exclusive=False):
+        """
+        Assigns a single role to a user with no checks from a category of roles
+        """
+        member = ctx.message.author
+        roles = self.roles[category].split(", ")
+
+        # normalize user input to title case
+        requested_role = requested_role.title()
+        role = utils.get(ctx.guild.roles, name=requested_role)
+
+        if (requested_role not in roles):
```

Comment:
ok now im getting pedantic but theres a lot of redundant parens in this diff
im fine with merging without changing these and ill do a cleanup later